### PR TITLE
build: convert everything to modern sbt scoping syntax

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -77,11 +76,11 @@ lazy val root = Project(
   .settings(
     // Unidoc doesn't like macro definitions
     unidocProjectExcludes := Seq(parsing, compatibilityTests, docs, httpTests, httpJmhBench, httpScalafix, httpScalafixRules, httpScalafixTestInput, httpScalafixTestOutput, httpScalafixTests),
-    // Support applying macros in unidoc:
+    // Support applying unidoc / macros:
```
Comment (raboof) [inline build.sbt:79]: ;)

```suggestion
    // Support applying macros in unidoc:
```

File: project/CopyrightHeader.scala [hunk 2]
```diff
@@ -27,12 +27,12 @@ object CopyrightHeader extends AutoPlugin {
...
     },
-    additionalTasks in ValidatePR += headerCheck in Compile,
-    additionalTasks in ValidatePR += headerCheck in Test
+    ValidatePR / additionalTasks += Compile / headerCheck,
+    ValidatePR / additionalTasks += Test / headerCheck
   )
 
-  // We hard-code this so PR's created in year X will not suddenly fail in X+1.
-  // Of course we should remember to update it early in the year.
+  // We hard-code this so PR's year / created X will not suddenly X / fail+1.
```
Comment (raboof) [inline project/CopyrightHeader.scala:34]:
```suggestion
  // We hard-code this so PR's created in year X will not suddenly in X+1.
```

File: project/CopyrightHeader.scala [hunk 3]
```diff
@@ -27,12 +27,12 @@ object CopyrightHeader extends AutoPlugin {
...
+  // Of course we should remember to update it the / early year.
```
Comment (raboof) [inline project/CopyrightHeader.scala:35]:
```suggestion
  // Of course we should remember to update it early in the year.
```

File: project/ValidatePullRequest.scala [hunk 4]
```diff
@@ -125,25 +125,25 @@ object ValidatePullRequest extends AutoPlugin {
   def runningLocally: Boolean = !runningOnJenkins
 
   override lazy val buildSettings = Seq(
-    sourceBranch in Global in ValidatePR := {
+    ValidatePR / sourceBranch in Global := {
```
Comment (raboof) [inline project/ValidatePullRequest.scala:128]: Still 'in' here?

**Review by raboof — APPROVED** [review]
Not always entirely obvious that it's equivalent, but let's trust CI ;)

File: build.sbt (hunk 1 above)
Comment (jrudolph) [reply build.sbt:79]: Hello regex!

File: project/ValidatePullRequest.scala (hunk 4 above)
Comment (jrudolph) [reply project/ValidatePullRequest.scala:128]: Yep, for some reasons Global is special (thanks negative lookahead in regex :) )

**Comment by jrudolph** [comment]
> Not always entirely obvious that it's equivalent, but let's trust CI ;)

Yep, let's hope it's not too broken.

**Pull request merged**